e2e/regressions/build.go

```diff
@@ -56,6 +57,37 @@ func (c *regressionsTests) issue4203(t *testing.T) {
 	)
 }
 
+// This test will build a sandbox, as a non-root user from a dockerhub image
+// that contains a single folder and file with `000` permissions.
+// To verify we force files to be accessible / moveable / removable by the user
+// we check for `700` and `400` permissions on the folder and file respectively.
+func (c *regressionsTests) issue4517(t *testing.T) {
+	sandbox := filepath.Join(c.env.TestDir, "issue_4203")
+
+	c.env.RunSingularity(
+		t,
+		e2e.WithPrivileges(false),
+		e2e.WithCommand("build"),
+		e2e.WithArgs("--sandbox", sandbox, "docker://sylabsio/issue4517"),
+		e2e.PostRun(func(t *testing.T) {
+			defer os.RemoveAll(sandbox)
```

Comment:
for debugging purposes, I would keep this directory in case of failure.

Also, I would capture any returned errors and log them if not nil.

So:

```
if t.Failed() {
    t.Logf("Test %s failed, not removing directory %s", t.Name(), sandbox)
    return
}

defer func() {
    err := os.RemoveAll(sandbox)
    if err != nil {
        t.Logf("Cannot remove sandbox directory: %#v", err)
    }
}()
```